### PR TITLE
Improvements to the handling of Gcode files

### PR DESCRIPTION
**Pull request by Wackerbarth**
These commits improve the way that gcode files are managed and reported.

It suppresses the listing of hidden artifact files in the SD directory
It rebases and corrects the changes from PR #145

### CODE REVIEW

File: redeem/gcodes/M2x.py [hunk 1]
```diff
@@ -116,8 +112,9 @@ def execute(self, g):
     self.printer.send_message(g.prot, "Begin file list:")
     for root, directories, filenames in os.walk(list_location):
       for filename in filenames:
-        file_byte_count = os.stat(root + os.sep + filename).st_size
-        self.printer.send_message(g.prot, "{}/{} {}".format(device_id, filename, file_byte_count))
+        if not filename.startswith('.'):
```
Comment (goeland86) [inline redeem/gcodes/M2x.py]: while a nice touch, OctoPrint tends to filter out all non-gcode files already.

I'd argue there may be cases where a user wants to see all the files present in the print folder though, even the ones that OctoPrint is filtering out.

File: redeem/gcodes/M2x.py [hunk 2]
```diff
@@ -399,19 +369,16 @@ def get_formatted_description(self):
...
     line_position, byte_position = self.printer.sd_card_manager.get_position()
     size_lines, size_bytes = self.printer.sd_card_manager.get_file_size()
     file_name = self.printer.sd_card_manager.get_file_name()
-
     # message to inform that we have completed the print
-    active = self.printer.sd_card_manager.get_status()
-    if active:
-      message = "SD printing byte {}/{}".format(byte_position, size_bytes)
-    else:
+    if file_name is None:
```
Comment (goeland86) [inline redeem/gcodes/M2x.py]: Isn't this set of changes already present in #145 ? Why are we not merging that, then you can build your changes on top of that?

Makes no sense that you're forcefully deprecating my PR while doing this. It's not the way I was instructed to use Git.

File: redeem/gcodes/M2x.py (hunk 1 above)
Comment (Wackerbarth) [reply redeem/gcodes/M2x.py]: Well, OP doesn't filter those out. Instead it clutters the left side of my screen with entries as if I could select them to print.

File: redeem/gcodes/M2x.py (hunk 2 above)
Comment (Wackerbarth) [reply redeem/gcodes/M2x.py]: Your PR is not mergeable. And, I have also corrected the code.

Comment (Wackerbarth) [reply redeem/gcodes/M2x.py]: If you wish to adopt the revised code and present it as a fast-forward merge to the present head, then go ahead and do so. As I told @ThatWileyGuy, I don't care about PRs. To me they just get in the way. It is the individual commits that matter. And this one is still attributed to you.

Comment (Wackerbarth) [reply redeem/gcodes/M2x.py]: And, BTW, "building on" to correct something is not a good practice. If it ain't right, it's not ready to be merged.

**Comment by goeland86** [comment]
Uh, ok, I need to look at my fork's code I think. Haven't had time to worry
much about it this week, too much happening in Oslo.

Give me a couple of days to get home and settled in a bit, and I'll work it
out.

**Comment by Wackerbarth** [comment]
I think that @ThatWileyGuy and I want to get this settled. And without this item, Redeem doesn't work with OP 1.3.10.  So, therefore, the build would be broken.

**Comment by goeland86** [comment]
Then merge it, by all means. I'm not going to be available to do much for
the next 48 hours - I will sort out my branch. You can close #145 in the
meantime.

**Pull request merged**